### DISCUSSION TITLE
RXC can't find a sub-function (right)

### DISCUSSION THREAD

**Discussion by Peter-Jacob** [adesutherland/CREXX#158] [Q&A] [answered]
In my new function X2D an internally called right function is not recognized anymore, although there is a prototype statement defined. I did a re-build this morning because of the new (IREM instruction). I looked at it for a while and can't find any error, maybe you have an idea:

C:\Users\PeterJ\CLionProjects\CREXX_F030\cmake-build-debug\compiler\rxc.exe x2d
Error @ 13:22 - #NOT_A_FUNCTION, "right(hex,slen,\'0\')"
1 error(s) in source file

```
/* rexx */
options levelb
/* X2d(hex-string)  returns decimal number of hex string */

x2d: procedure = .int
  arg hex = .string, slen=-1
  hlen=0
  char=0
  dec=0
  sign=0
  if slen=0 then return 0      /* no hex input return 0    */
  assembler strlen hlen,hex    /* get length of hex string */
  if slen>0 then hex=right(hex,slen,'0')  /* if length parm set, treat hex as signed integer */
  else slen=hlen                          /* else it is an unsignded integer

  do i=0 to slen-1                     /* loop through hex string */
     assembler strchar char,hex,i      /* fetch one byte          */
     if char>=97 & char<=102 then do   /* treat a to f has hex numbers */
        char=char-87
     end
     else do                           /* else treat A to F has hex numbers */
       if char>=65 & char <=70 then do
          char=char-55
       end
       else do
         if char>=48 & char <=57 then do /* else translate num chars     */
            char=char-48
         end
         else do                        /* else hex string contains invalid chars */
            say 'invalid hex code 'char
            return -1
         end
       end
     end
  /* if it is first byte then check if sign is set, and convert number */
     if i=0 then do
        if char>=8 & slen>0 then do /* does it contain sign? */
           sign=1           /* keep sign       */
           char=char-8      /* switch off sign */
        end
     end

     dec=dec*16+char        /* cumulate fetched char in result integer */
  end

/* hex string has been processed */
  if sign=1 then do        /* was sign set? */
     char=8                /* build complement */
     do i=1 to slen-1
        char=char*16
     end
     dec=dec-char          /* create integer out of complement */
  end
return dec

/* Right() Procedure */
  right: procedure = .string
         arg string = .string, length1 = .int, pad = '0'
```

**Comment by adesutherland** [comment, marked as answer]
Line 14

    else slen=hlen                          /* else it is an unsignded integer

is missing an end-comment (*/) and the compiler is not issuing a error about an unclosed comment.

**Comment by Peter-Jacob** [comment]
this is embarrassing :-(. Maybe I should give up programming and play Bingo instead.

**Reply by adesutherland** [reply]
Nah - if the compiler doesn't give the right errors what is a programmer meant to do!

**Comment by rvjansen** [comment]
Only if it is the kind of bingo that uncovers a lot of bugs!

René.

**Comment by Peter-Jacob** [comment]
Alternatively, I could stop writing inline documentation ...

**Reply by adesutherland** [reply]
Or dare we risk anger and implement line comments? I am thinking of /--

**Reply by Peter-Jacob** [reply]
I am fine with ` /--`

**Reply by Peter-Jacob** [reply]
anger with whom? I can take a lot without falling apart.

**Reply by adesutherland** [reply]
Oh - I think that Rene mentioned that there has been a lot of debate about line comments in REXX, and an acceptable syntax

**Reply by Peter-Jacob** [reply]
ah, I see you mean people who like endless discussions but never contribute. 
what about implementing it as a level b only or leave it undocumented in level b/c?

**Comment by rvjansen** [comment]
I love line comments. Two dashes, like NetRexx, would be ideal.

René.

**Reply by Peter-Jacob** [reply]
-- also nice, one char less then /--

**Reply by Peter-Jacob** [reply]
you are the PM make a  decision!

**Reply by adesutherland** [reply]
Increment and decrement operators ... ?

**Comment by rvjansen** [comment]
Yes, that is a thing. NetRexx does not have them, I am not missing them.

**Reply by Peter-Jacob** [reply]
oh yes, -- ++ operators. I'd like them ...

**Comment by rvjansen** [comment]
I would not rule out that I would use them if they are available. But for compatibility - I checked ooRexx and Regina, they all have double dash as a comment. Would it be hard to have them in level B as a decrement operator *and* as a line comment? (the dashes need to decrement something, a variable which is not there if used as a line comment).

**Reply by adesutherland** [reply]
> Would it be hard to have them in level B as a decrement operator and as a line comment?

Impossible given how flexible REXX syntax is! :-)

Even /-- is troublesome, e.g. b = a/--x is a valid expression if we have a -- operator.

So I am thinking the answer may be ---

Anyway - no rush to decide 

In classic rexx adding -- is feasible
